HRIS.Services/Services/WorkExperienceService.cs

```diff
@@ -0,0 +1,76 @@
+﻿using HRIS.Models;
+using HRIS.Services.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using RR.UnitOfWork;
+using RR.UnitOfWork.Entities.HRIS;
+
+namespace HRIS.Services.Services;
+
+public class WorkExperienceService : IWorkExperienceService
+{
+    private readonly IUnitOfWork _db;
+    private readonly IErrorLoggingService _errorLoggingService;
+
+    public WorkExperienceService(IUnitOfWork db, IErrorLoggingService errorLoggingService)
+    {
+        _db = db;
+        _errorLoggingService = errorLoggingService;
+    }
+
+    public async Task<bool> CheckIfExists(WorkExperienceDto workExperience)
+    {
+        var exists =  await _db.WorkExperience.Any(x => x.Id == workExperience.Id);
+
+        return exists;
+    }
+
+    public async Task<WorkExperienceDto> Save(WorkExperienceDto workExperience)
+    {
+        var exists = await CheckIfExists(workExperience);
+
+        if (exists)
+        {
+            var exception = new Exception("Work experience already exists");
+            throw _errorLoggingService.LogException(exception);
```

Comment:
You can pass the exception straight into the method because you're not using it elsewhere

Reply:
ok cool